ruleguard/go.mod

```diff
@@ -0,0 +1,5 @@
+module ruleguard
+
+go 1.21
```

Comment:
Out of scope but I wonder if it's worth thinking about adding `toolchain 1.21` to our code

Reply:
What does that give us?

Reply:
The compiler will download a specific version of go toolchain (which is possible since go 1.21), which would make it possible to manage the versions of go used in only one place `go.mod` (vs having to manage in github workflows, Jenkinsfile, jenkins itself.

It also would be useful if we update go on a different branch, since local development would pick up that version.

Reply:
Ok, this sounds interesting but not in the context of this PR. Feel free to file a ticket.